programs/bpf_loader/src/syscalls.rs

```diff
@@ -2107,136 +2305,1723 @@ fn call<'a>(
+}
 
-    #[test]
-    fn test_translate() {
-        const START: u64 = 100;
-        const LENGTH: u64 = 1000;
-        let data = vec![0u8; LENGTH as usize];
-        let addr = data.as_ptr() as u64;
-        let memory_mapping = MemoryMapping::new::<UserError>(
-            vec![MemoryRegion::new_from_slice(&data, START, 0, false)],
-            &DEFAULT_CONFIG,
-        )
-        .unwrap();
-
-        let cases = vec![
-            (true, START, 0, addr),
-            (true, START, 1, addr),
-            (true, START, LENGTH, addr),
-            (true, START + 1, LENGTH - 1, addr + 1),
-            (false, START + 1, LENGTH, 0),
-            (true, START + LENGTH - 1, 1, addr + LENGTH - 1),
-            (true, START + LENGTH, 0, addr + LENGTH),
-            (false, START + LENGTH, 1, 0),
-            (false, START, LENGTH + 1, 0),
-            (false, 0, 0, 0),
-            (false, 0, 1, 0),
-            (false, START - 1, 0, 0),
-            (false, START - 1, 1, 0),
-            (true, START + LENGTH / 2, LENGTH / 2, addr + LENGTH / 2),
-        ];
-        for (ok, start, length, value) in cases {
-            if ok {
-                assert_eq!(
-                    translate(&memory_mapping, AccessType::Load, start, length).unwrap(),
-                    value
-                )
+/// Transposes a FfiBignumber array of bytes to a bn::BigNum
+fn ffibignumber_to_bignum(ffi: &FfiBigNumber, ffi_data: &[u8]) -> Result<BigNum, SyscallError> {
+    let ffi_bn = BigNum::from_slice(ffi_data);
+    match ffi_bn {
+        Ok(mut bn) => {
+            bn.set_negative(ffi.is_negative);
+            Ok(bn)
+        }
+        Err(es) => {
+            if let Some(reason) = es.errors()[0].reason() {
+                Err(SyscallError::BigNumberOpenSSLError(reason.to_string()))
             } else {
-                assert!(translate(&memory_mapping, AccessType::Load, start, length).is_err())
+                Err(SyscallError::BigNumberOpenSSLErrorUnknown)
             }
         }
     }
+}
 
-    #[test]
-    fn test_translate_type() {
-        // Pubkey
-        let pubkey = solana_sdk::pubkey::new_rand();
-        let addr = &pubkey as *const _ as u64;
-        let memory_mapping = MemoryMapping::new::<UserError>(
-            vec![MemoryRegion {
-                host_addr: addr,
-                vm_addr: 100,
-                len: std::mem::size_of::<Pubkey>() as u64,
-                vm_gap_shift: 63,
-                is_writable: false,
-            }],
-            &DEFAULT_CONFIG,
-        )
-        .unwrap();
-        let translated_pubkey =
-            translate_type::<Pubkey>(&memory_mapping, 100, &bpf_loader::id(), true).unwrap();
-        assert_eq!(pubkey, *translated_pubkey);
-
-        // Instruction
-        let instruction = Instruction::new_with_bincode(
-            solana_sdk::pubkey::new_rand(),
-            &"foobar",
-            vec![AccountMeta::new(solana_sdk::pubkey::new_rand(), false)],
+/// BIGNUM sol_bignum_from_u32
```

Comment:
Why not `sol_bignum_from_u64`?

Reply:
Not supported available through OpenSSL BigNum